SessionMessagingKit/Utilities/SessionPro/SessionProState+BottomSheet.swift

```diff
@@ -0,0 +1,389 @@
+                guard let self = self else { return }
+                
+                self.state.updateTableData(updatedState.sections(viewModel: self))
+                self.internalState = updatedState
+            }
+    }
+    
+    // MARK: - Config
+    
+    public enum Section: SessionListScreenContent.ListSection {
+        case logoWithPro
+        case proFeatures
+        
+        public var title: String? {
+            switch self {
+                case .proFeatures: return "proBetaFeatures".put(key: "pro", value: Constants.pro).localized()
+                default: return nil
+            }
+        }
+        
+        public var style: SessionListScreenContent.ListSectionStyle {
+            switch self {
+                case .proFeatures: return .titleNoBackgroundContent
+                default: return .none
+            }
+        }
+        
+        public var divider: Bool { false }
+        public var footer: String? { return nil }
+    }
+    
+    public enum ListItem: Differentiable {
+        case logoWithPro
+        case continueButton
+
+        case longerMessages
+        case unlimitedPins
+        case animatedDisplayPictures
+        case badges
+        case plusLoadsMore
+    }
+    
+    // MARK: - Content
+    
+    public struct ViewModelState: ObservableKeyProvider {
+        let currentProPlanState: SessionProPlanState
+        let loadingState: SessionProLoadingState
+        
+        @MainActor public func sections(viewModel: SessionProBottomSheetViewModel) -> [SectionModel] {
+            SessionProBottomSheetViewModel.sections(
+                state: self,
+                viewModel: viewModel
+            )
+        }
+        
+        public let observedKeys: Set<ObservableKey> = [
+            .feature(.mockCurrentUserSessionProState),          // TODO: real data from libSession
+            .feature(.mockCurrentUserSessionProLoadingState)    // TODO: real loading status
+        ]
+        
+        static func initialState() -> ViewModelState {
+            return ViewModelState(
+                currentProPlanState: .none,
+                loadingState: .loading
+            )
+        }
+    }
+    
+    @Sendable private static func queryState(
+        previousState: ViewModelState,
+        events: [ObservedEvent],
+        isInitialQuery: Bool,
+        using dependencies: Dependencies
+    ) async -> ViewModelState {
+        var currentProPlanState: SessionProPlanState = previousState.currentProPlanState
+        var loadingState: SessionProLoadingState = previousState.loadingState
+        
+        currentProPlanState = dependencies[singleton: .sessionProState].sessionProStateSubject.value
+        loadingState = dependencies[feature: .mockCurrentUserSessionProLoadingState]
+        
+        return ViewModelState(
+            currentProPlanState: currentProPlanState,
+            loadingState: loadingState
+        )
+    }
+    
+    private static func sections(
```

Comment:
Were there many differences between this and the `SessionProSettingsViewModel`? (just thinking that it's a lot of potentially duplicate code to maintain)